src/transformer_util.ts

```diff
@@ -190,19 +189,51 @@ export function createMultiLineComment(original: ts.Node, text: string) {
  * to do. By default, tsickle does not report any warnings to the caller, and warnings are hidden
  * behind a debug flag, as warnings are only for tsickle to debug itself.
  */
-export function debugWarn(
-    host: {logWarning ? (d: ts.Diagnostic) : void}, context: ts.Node, messageText: string) {
+export function reportDebugWarning(
+    host: {logWarning ? (d: ts.Diagnostic) : void}, node: ts.Node, messageText: string) {
   if (!host.logWarning) return;
-  // Use a ts.Diagnosic so that the warning includes context and file offets.
-  const diagnostic: ts.Diagnostic = {
-    file: context.getSourceFile(),
-    start: context.getStart(),
-    length: context.getEnd() - context.getStart(),
+  host.logWarning(createDiagnostic(node, ts.DiagnosticCategory.Warning, messageText));
+}
+
+/**
+ * Creates and reports a diagnostic by adding it to the given array.
+ *
+ * This is used for erros and warnings in tsickle's input. Emit errors (the default) if tsickle
+ * cannot emit a correct result given the input. Emit warnings for questionable input if there's a
+ * good chance that the output will work.
+ *
+ * For typical tsickle users, errors are always reported and break the compilation operation,
```

Comment:
typical = google3?  This user seemed to be experiencing warnings as errors, though I'm not sure what tool they were using. https://github.com/angular/tsickle/issues/863

Reply:
Yeah, typical = google3. I have no idea about that other thing.